packages/itwin/measure-tools/src/measurements/PerpendicularDistanceMeasurement.ts

```diff
@@ -0,0 +1,141 @@
+    return this._secondaryLine ?? [];
+  }
+  public set secondaryLine(l: Point3d[]) {
+    this._secondaryLine = l;
+  }
+  public get toolName(): string {
+    return this._toolName ?? "";
+  }
+  public set toolName(t: string) {
+    this._toolName = t;
+  }
+
+  public static override create(start: Point3d, end: Point3d, viewType?: string) {
+    // Don't ned to serialize the points, will just work as is
+    const measurement = new PerpendicularDistanceMeasurement({ startPoint: start, endPoint: end });
+    if (viewType) {
+      measurement.viewTarget.include(viewType);
+    }
+
+    return measurement;
+  }
+
+  public override decorate(context: DecorateContext): void {
+    super.decorate(context);
+
+    if (this._secondaryLine && this._secondaryLine.length > 0) {
+      const secondaryLine = context.createGraphicBuilder(GraphicType.WorldOverlay, undefined, this.getSnapId());
+      secondaryLine.setSymbology(ColorDef.white, ColorDef.black, 1, LinePixels.Code5);
+      secondaryLine.addLineString(this._secondaryLine);
+      context.addDecorationFromBuilder(secondaryLine);
+    }
+  }
+
+  protected override async getDataForMeasurementWidgetInternal(): Promise<MeasurementWidgetData> {
+    const toolName = this._toolName ?? "";
+    const lengthSpec = await IModelApp.quantityFormatter.getFormatterSpecByQuantityType(QuantityType.LengthEngineering);
+
+    const distance = this.worldScale * this.startPointRef.distance(this.endPointRef);
+    const run =
+      this.drawingMetadata?.worldScale !== undefined
+        ? this.worldScale * Math.abs(this.endPointRef.x - this.startPointRef.x)
+        : this.startPointRef.distanceXY(this.endPointRef);
+    const rise =
+      this.drawingMetadata?.worldScale !== undefined
+        ? this.worldScale * (this.endPointRef.y - this.startPointRef.y)
+        : this.endPointRef.z - this.startPointRef.z;
+    const slope = 0.0 < run ? (100 * rise) / run : 0.0;
+    const dx = Math.abs(this.endPointRef.x - this.startPointRef.x);
+    const dy = Math.abs(this.endPointRef.y - this.startPointRef.y);
+
+    const adjustedStart = this.adjustPointForGlobalOrigin(this.startPointRef);
+    const adjustedEnd = this.adjustPointForGlobalOrigin(this.endPointRef);
+
+    const fDistance = IModelApp.quantityFormatter.formatQuantity(distance, lengthSpec);
+    const fStartCoords = FormatterUtils.formatCoordinatesImmediate(adjustedStart);
+    const fEndCoords = FormatterUtils.formatCoordinatesImmediate(adjustedEnd);
+    const fSlope = FormatterUtils.formatSlope(slope, true);
+    const fRun = IModelApp.quantityFormatter.formatQuantity(run, lengthSpec);
+    const fDeltaX = IModelApp.quantityFormatter.formatQuantity(dx, lengthSpec);
+    const fDeltaY = IModelApp.quantityFormatter.formatQuantity(dy, lengthSpec);
+    const fRise = IModelApp.quantityFormatter.formatQuantity(rise, lengthSpec);
+
+    let title = `${toolName} ${MeasureTools.localization.getLocalizedString("MeasureTools:Measurements.measurement")}`;
```

Comment:
I know there are some places that do that already (and it's incorrect!), but it is bad practice to concatenate localized strings. Things may be written wildly differently in another language.
Consider writing `Height measurement` or `Width measurement` with separate localization keys, and get the right one based off some property in this measurement object (see my prior comment).

Reply:
Fixed localization